lib/helper/browser.go

```diff
@@ -223,16 +229,39 @@ func OpenBrowserRedirect(target string, typeID uint) error {
 	// generate the federation link
 	federationLink := fmt.Sprintf("%s%s", logoutURL, encodedUrl)
 
-	// open the browser
-	switch runtime.GOOS {
-	case "linux":
-		err = exec.Command("xdg-open", federationLink).Start()
-	case "windows":
-		err = exec.Command("rundll32", "url.dll,FileProtocolHandler", federationLink).Start()
-	case "darwin":
-		err = exec.Command("open", federationLink).Start()
-	default:
-		err = fmt.Errorf("unsupported platform")
+	if config.FirefoxContainers {
+		fmt.Printf("Opening in Firefox Container\n")
+		federationLink = fmt.Sprintf("ext+granted-containers:name=%s&url=%s", name, url.QueryEscape(federationLink))
```

Comment:
Is it worth just using the `target` here instead of the `federationLink` since we're not contending with existing sessions? The target is just the standard federation URL before we wrap it into a logout URL.

Reply:
Indeed it is. Cleaned this up a bit.